compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -784,8 +788,42 @@ impl<'a> TraitDef<'a> {
         // Create the type of `self`.
         let path = cx.path_all(self.span, false, vec![type_ident], self_params);
         let self_type = cx.ty_path(path);
+        let rustc_const_unstable =
+            cx.path_ident(self.span, Ident::new(sym::rustc_const_unstable, self.span));
+
+        let mut attrs = thin_vec![cx.attr_word(sym::automatically_derived, self.span),];
+
+        if self.is_const && self.is_staged_api_crate {
```

Comment:
There's clearly something interesting going on here, please add a comment explaining what.

Reply:
I'm fine with this being a temporary solution. To me, we should change `#[derive_const(PartialEq)]` to something like `#[derive(PartialEq(const))]` (syntax to-be-bikeshedded), then we can have `#[derive(PartialEq(const, unstaable))]` in all standard library crates.

Reply:
How does it know which feature gate to use...?
We;re fairly close to actually enforcing feature stability on impls (https://github.com/rust-lang/rust/pull/140399) so this question matters soon, IIUC.